### PR TITLE
Build orchestration updates

### PR DESCRIPTION
**Pull request by vatsan-madhavan**
Today, the assemblies built out of [dotnet/wpf](https://github.com/dotnet/wpf) are packaged into a transport NuGet package (`Microsoft.DotNet.Wpf.GitHub`). This package is consumed by an internal repo that builds native DLL's, and produces another transport package `Microsoft.DotNet.Wpf.DncEng`. 

Both of these transport packages are in eventually consumed by [dotnet/windowsdesktop](https://github.com/dotnet/windowsdesktop) (alongwith a transport package from WinForms) to build the WindowsDesktop shared runtime. 

These updates to build targets will enable us to build a single transport package - `Microsoft.DotNet.Wpf.GitHub` - to flow into `dotnet/windowsdesktop`. This repo will use `Microsoft.DotNet.Wpf.DncEng` as an input (which contains native DLL's) and will produce a single transport package that contains all relevant WPF binaries (including those that came from `Microsoft.DotNet.Wpf.DncEng`). 

These changes along **do not** institute changes to the overall build orchestration, and will not make effective the intent described above. These changes are intended to enable the goals outlined here shortly. Additional changes will likely be needed before the build orchestration can be further streamlined like this: 

Now:
```
             +-----------------------------------------------------------------------------------------------------+
             |                                                                                                     |
             |                                                                                                     |
             |                                                                                                     |
             |                                                                                                     |
             |                                                                                                     |
             |                        +-----------------+    Microsoft.DotNet.Wpf.GitHub                           |
             |                 +----->+                 +-------------------------------------------------+ +------v---------------------+
+------------+-----+           |      |   WPF           |                                                 | |   dotnet/windowsdesktop    |
|                  |           |      +-------+---------+                                                 | |                            |        Microsoft.WindowsDesktop.App
|  WinForms        +-----------+              |                                                           +->     WindowsDesktop         +------>
|                  |           |              |                                                             |      composition           |
+------------------+           +-----+       Microsoft.DotNet.Wpf.GitHub                                    |                            |
                                     |        |                                                             |                            |
                                     |        |                                                             +--------------^-------------+
                                     |        |                     +--------------------------------+                     |
                                     |        +--------------------->       dotnet-wpf-int           |                     |
                                     Microsoft.Private.Winforms----v+                                |                     |
                                                                    |                                |                     |
                                                                    |   Native binaries built here   |                     |
                                                                    |                                |                     |
                                                                    |                                +-----------Microsoft.DotNet.Wpf.DncEng
                                                                    |                                |
                                                                    |                                |
                                                                    |                                |
                                                                    +--------------------------------+


```

Eventually:

```
                                                     Microsoft.Private.Winforms
                +-----------------------------------------------------------------------------------------------+
                |                                                                                               |
                |                                                                                               |
                |                                                                                               |
                |                                                                                               |
+---------------+-----------+                     +-----------------------------+                 +-------------v------------------+
|                           |                     |                             |                 |      dotnet/windowsdesktop     |
|                           |                     |                             |                 |                                |
|    WinForms               +--------------------->        WPF                  +-Microsoft.DotNet.Wpf.GitHub                      |
|                           |                     |                             |                <|                                |
|                           |                     |                             |                 |                                |
+---------------------------+                     |                             +--------+        |                                |
                                                  +--------------+--------------+        |        +--------------------------------+
                                                                 ^                       |
                                                                 |                       |
                                                                 |                       |
                                                                 |                       |
                                                            Microsoft.DotNet.Wpf.DncEng  | reverse flow:
                                                                 |                       |   1x/day toolset only
                                                                 |                       |   update (build props/targets)
                                                                 |                       |
                                                                 |                       |
                                                                 |                       |
                                                                 |                       |
                                                   +-------------+---------------+       |
                                                   |     dotnet-wpf-int          |       |
                                                   |                             |       |
                                                   |                             |       |
                                                   |                             <-------+
                                                   |                             |
                                                   |                             |
                                                   +-----------------------------+
```

### CODE REVIEW

File: eng/WpfArcadeSdk/tools/Wpf.Cpp.targets [hunk 1]
```diff
@@ -436,6 +436,145 @@ using namespace System::Runtime::Versioning;
...
+        </Text>
+      </CppSupportProject>
+    </ItemGroup>
+
+    <MakeDir Condition="!Exists('$(IntermediateOutputPath)CppSupport\')"
+             Directories="$(IntermediateOutputPath)CppSupport\" />
+
+    <Delete Files="$(CppHelperProject)" Condition="Exists('$(CppHelperProject)')" />
+
+    <WriteLinesToFile Lines="%(CppSupportProject.Text)"
```
Comment (lindexi) [inline eng/WpfArcadeSdk/tools/Wpf.Cpp.targets]: Maybe we can use `MSBuild Escape` that we can use `;` and `@` etc.

See https://docs.microsoft.com/en-us/visualstudio/msbuild/how-to-escape-special-characters-in-msbuild?view=vs-2019 and https://blog.lindexi.com/post/Roslyn-how-to-use-WriteLinesToFile-to-write-the-semicolons-to-file.html

Here is the example from my blog

```
    <PropertyGroup>
      <SomeThing>
        <![CDATA[
            Console.WriteLine("I am lindexi");
       ]]>
      </SomeThing>
    </PropertyGroup>
    <WriteLinesToFile File="lindexi.cs" Lines="$([MSBuild]::Escape($(SomeThing)))" Overwrite="true"></WriteLinesToFile>
```

File: eng/WpfArcadeSdk/tools/Wpf.Cpp.targets [hunk 2]
```diff
@@ -436,6 +436,145 @@ using namespace System::Runtime::Versioning;
...
+    </PropertyGroup>
+    <!-- Restore, then IdentifyNuGetReferences -->
+    <MSBuild Projects="@(ProjectToBuild)" UnloadProjectsOnCompletion="true"
+             Targets="Restore"
+             Properties="TargetFramework=$(TfmForBuild);Platform=$(Architecture);RuntimeIdentifier=win-$(Architecture);IsRestoring=true" />
+    
+    <MSBuild Projects="@(ProjectToBuild)"
+         Targets="IdentifyNuGetReferences"
+         Properties="TargetFramework=$(TfmForBuild);Platform=$(Architecture);RuntimeIdentifier=win-$(Architecture)">
+      <Output  TaskParameter="TargetOutputs" ItemName="NuGetReferencesForCpp" />
```
Comment (lindexi) [inline eng/WpfArcadeSdk/tools/Wpf.Cpp.targets]:
```suggestion
      <Output TaskParameter="TargetOutputs" ItemName="NuGetReferencesForCpp" />
```

File: eng/WpfArcadeSdk/tools/Wpf.Cpp.targets (hunk 1 above)
Comment (vatsan-madhavan) [reply eng/WpfArcadeSdk/tools/Wpf.Cpp.targets]: This is a tested pattern, so I don't want to change this. Most of this pattern will go away shortly when we switch to adopting vcxproj support in Dev16.4 for C++/CLI. What remains will be minimal and easy enough to maintain (and is not intended to be maintained by the community at large- its mainly targeted at a couple of maintainers in the WPF team).

**Comment by vatsan-madhavan** [comment]
@rladuca can you take a look please?

File: eng/WpfArcadeSdk/tools/WpfProjectReference.targets [hunk 3]
```diff
@@ -42,11 +52,30 @@
...
-    <WpfProjectPath Include="PresentationNative_cor3" ProjectPath="$(WpfSourceDir)PresentationNative\dll\PresentationNative.vcxproj" />
-    <WpfProjectPath Include="WpfGfx_cor3" ProjectPath="@(WpfSourceDir)WpfGfx\core\dll\WpfGfx.vcxproj" />
-    <WpfProjectPath Include="PenImc_cor3" ProjectPath="$(WpfSourceDir)PenImc\dll\PenImc.vcxproj" />
+
+    <WpfProjectPath Include="PresentationNative_cor3" ProjectPath="$(WpfSourceDir)PresentationNative\dll\PresentationNative.vcxproj">
+      <ReferenceOutputAssembly>false</ReferenceOutputAssembly>
+      <OutputItemType>Content</OutputItemType>
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </WpfProjectPath>
+    <WpfProjectPath Include="WpfGfx_cor3" ProjectPath="@(WpfSourceDir)WpfGfx\core\dll\WpfGfx.vcxproj">
```
Comment (rladuca) [inline eng/WpfArcadeSdk/tools/WpfProjectReference.targets]: $(WpfSourceDir)

**Pull request merged**